components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java

```diff
@@ -1206,6 +1206,108 @@ public void removeSubscription(APIIdentifier identifier, int applicationId)
 		}
 	}
 
+	/**
+	 * Moves subscriptions of one app to another app
+	 *
+	 * @param fromIdentifier subscriptions of this app
+	 * @param toIdentifier   will be moved into this app
+	 * @return number of subscriptions moved
+	 * @throws AppManagementException
+	 */
+	public int moveSubscriptions(APIIdentifier fromIdentifier, APIIdentifier toIdentifier)
+			throws AppManagementException {
+
```

Comment:
Now `moveSubscriptions` method is broken into to smaller methods, see commit 07d4afd84e007233c5a9ee791d8288c5bffb3cbf of PR #153